### PR TITLE
Improve scalability of uan_interfaces role and make it idempotent when NMN bonding is active

### PR DESCRIPTION
**Pull request by keopp-hpe**
## Summary and Scope

This PR improves the scaling of the 'uan_interfaces' role.  It also cleans up interface files when networking has changed.  The role is now idempotent when NMN bonding is active.  Idempotent operation involved making 'uan_hardening' NMN bonding aware.

## Issues and Related PRs

* Resolves [CASMUSER-3186](https://jira-pro.it.hpe.com:8443/browse/CASMUSER-3186)
* Resolves [CASMUSER-3107](https://jira-pro.it.hpe.com:8443/browse/CASMUSER-3107)

### Tested on:

  * `shandy` (CAN system)
  * `surtur` (CHN system)

### Test description:

Installed updated 'uan_interfaces' role and booted each system.  Also ran CFS to update configuration without a reboot.

## Risks and Mitigations

Low risk.

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [ ] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: ansible/roles/uan_interfaces/tasks/main.yml [hunk 1]
```diff
@@ -327,6 +319,80 @@
...
+
+    - name: "Check if ifcfg-{{ uan_chn_device }} exists"
+      stat:
+        path: "/etc/sysconfig/network/ifcfg-{{ uan_chn_device }}"
+      register: ifcfg_chn_dev
+
+    - block:
+      - name: Gather any CAN/CHN bond slave devices for removal from ifcfg-bond0, if not configured for CAN/CHN
+        slurp:
+          src: "/etc/sysconfig/network/ifcfg-bon0"
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply ansible/roles/uan_interfaces/tasks/main.yml:347]: Good eye!  bond0 is what it should be!

File: ansible/roles/uan_interfaces/tasks/main.yml [hunk 2]
```diff
@@ -343,3 +412,20 @@
...
   when:
     - ('Compute' not in group_names) or ('Compute' in group_names and uan_user_access_cfg == 'CHN')
+
+- name: "DBG: Get uan_interfaces exit time"
+  local_action:
+    module: shell
+    cmd: 'date "+%T"'
+  register: end_time
+
+- name: "DBG: Performance times"
```
Comment (alanm-hpe) [inline ansible/roles/uan_interfaces/tasks/main.yml:422]: I don't think all of the debug statements are necessary. With ARA, we can get a breakdown of the time spent in each task and a summary of the whole playbook.

image

File: ansible/roles/uan_interfaces/tasks/reload_networks.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+---
+- name: Reload networks to update default route 
+  shell: "systemctl reload wicked;sleep 5"
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply ansible/roles/uan_interfaces/tasks/reload_networks.yml]: Works fine without sleeps.

File: ansible/roles/uan_interfaces/tasks/main.yml (hunk 2 above)
Comment (keopp-hpe) [reply ansible/roles/uan_interfaces/tasks/main.yml]: Removed debug statements and we'll use ARA to get performance data.

**Review by alanm-hpe — APPROVED** [review]

**Pull request merged**